db/seeds.rb

```diff
@@ -58,542 +132,329 @@ def time_rand(from = 0.0, to = Time.now, length = 0, options = {})
+  res.due_date = res.due_date - factor
+  mark_checked_out res
+  mark_checked_in(res, res.equipment_model.category.max_checkout_length)
+  res.save(validate: false)
+end
 
+# rubocop:disable AbcSize
+def generate_reservation
+  count = 0
+  while count < MAX_TRIES do
+    res = Reservation.new
+    res.reserver_id = User.all.sample.id
+    res.equipment_model = EquipmentModel.all.sample
+    res.start_date = time_rand(Time.now, Time.now + FUTURE_RANGE).to_datetime
+
+    checkout_length = res.equipment_model.category.max_checkout_length
+    res.due_date = time_rand(res.start_date, res.start_date.next_week,
+                             checkout_length.days).to_datetime
+    res.notes = Faker::HipsterIpsum.paragraph(8)
+    res.notes_unsent = [true, false].sample
+    if rand < PAST_CHANCE
+      throw_into_past res 
+      return
+    end
+    try_again = false
+    begin
+      res.save!
+    rescue ActiveRecord::RecordInvalid
+      try_again = true
+    end
+    return unless try_again
+    count += 1
+  end
+end
+# rubocop:enable AbcSize
+
+def generate_objs(method, obj, n)
+  return if n == 0
+  progress = ProgressBar.create(format: PROGRESS_STR, total: n)
+  n.times.map do
+    progress.increment
+    send(method)
+  end
+  puts "\n#{n} #{obj.camelize} records successfully created!"
+end
 
 # START SCRIPT
 # ============
+#
 
-if ENV["minimal"]
-  puts "Minimal mode activated. Please wait..."
-end
 
+puts 'Minimal mode activated. Please wait...' if MINIMAL
+
+t1 = Time.now
+display_login_msg = false
 # check to see if a superuser exists
 if User.where('role = ?', 'superuser').empty?
+  display_login_msg = true
   User.destroy_all
-  unless ENV["minimal"]
-    puts 'We need to create an account for you first. Please enter the following info:'
-    puts 'First Name:'
-    first_name = STDIN.gets.chomp
-    puts 'Last Name:'
-    last_name = STDIN.gets.chomp
-    puts 'Phone #:'
-    phone = STDIN.gets.chomp
-    puts 'Email Address:'
-    email = STDIN.gets.chomp
-    puts 'Affiliation:'
-    affiliation = STDIN.gets.chomp
+  u = User.new
+  u.first_name = 'Donny'
+  u.last_name = 'Darko'
+  u.phone = '6666666666'
+  u.email = 'email@email.com'
+  u.affiliation = 'Your Mother'
+  u.role = 'superuser'
+  u.view_mode = 'superuser'
+  u.username = 'dummy'
+  u.save
+
+  if MINIMAL || SEMI
     if ENV['CAS_AUTH']
-      puts 'Username (i.e. NetID):'
-      username = STDIN.gets.chomp
+      printf 'CAS '
+      prompt_field(u, :username)
     else
-      username = email
-      puts 'Password:'
-      password = STDIN.gets.chomp
-      puts 'Confirm Password:'
-      password_confirmation = STDIN.gets.chomp
+      u.username = u.email
+      u.password = 'passw0rd'
+      u.password_confirmation = u.password
+      u.save
     end
   else
-    first_name = "Donny"
-    last_name = "Darko"
-    phone = "6666666666"
-    email = "email@email.com"
-    affiliation = "Your Mother"
+    puts 'We need to create an account for you first.' \
```

Comment:
You need to add a space after the period. My practice with multiline strings has generally been to leave the trailing space at the end of each line rather than stick it at the beginning of the following line, but I don't think it matters much stylistically.